common/client-core/gateways-storage/src/backend/fs_backend/mod.rs

```diff
@@ -134,18 +132,20 @@ impl GatewaysDetailsStore for OnDiskGatewaysDetails {
         Ok(())
     }
 
-    async fn upgrade_stored_remote_gateway_key(
+    async fn update_gateway_details(
         &self,
-        gateway_id: ed25519::PublicKey,
-        updated_key: &SharedSymmetricKey,
+        details: &GatewayRegistration,
     ) -> Result<(), Self::StorageError> {
-        self.manager
-            .update_remote_gateway_key(
-                &gateway_id.to_base58_string(),
-                None,
-                Some(updated_key.as_bytes()),
-            )
-            .await?;
+        match &details.details {
+            GatewayDetails::Remote(remote_details) => {
+                let raw_details = &remote_details.into();
+                self.manager
+                    .update_remote_gateway_details(raw_details)
+                    .await?;
+            }
+            GatewayDetails::Custom(_) => {}
```

Comment:
it feels a bit weird that even though we're supposedly updating gateway detaiils, there's absolutely nothing being done for `Custom` variant. imo the storage function should be made less generic and should only apply to the data we're actually changing

Reply:
Fixed